lib/src/revset.rs

```diff
@@ -1956,20 +1999,28 @@ pub fn evaluate_expression<'repo>(
             repo,
             &repo.view().public_heads().iter().cloned().collect_vec(),
         )),
-        RevsetExpression::Branches => {
+        RevsetExpression::Branches(needle) => {
             let mut commit_ids = vec![];
-            for branch_target in repo.view().branches().values() {
+            for (branch_name, branch_target) in repo.view().branches().iter() {
+                if !branch_name.contains(needle) {
+                    continue;
+                }
                 if let Some(local_target) = &branch_target.local_target {
                     commit_ids.extend(local_target.adds());
                 }
             }
             Ok(revset_for_commit_ids(repo, &commit_ids))
         }
-        RevsetExpression::RemoteBranches => {
+        RevsetExpression::RemoteBranches(branch_needle, remote_needle) => {
             let mut commit_ids = vec![];
-            for branch_target in repo.view().branches().values() {
-                for remote_target in branch_target.remote_targets.values() {
-                    commit_ids.extend(remote_target.adds());
+            for (branch_name, branch_target) in repo.view().branches().iter() {
+                if !branch_name.contains(branch_needle) {
+                    continue;
```

Comment:
Nit: `.iter()` isn't needed.

Alternatively, `.iter().filter()` can be used.

Reply:
Done.